pyccel/parser/semantic.py

```diff
@@ -1339,6 +1339,9 @@ def _ensure_target(self, rhs, d_lhs):
         if isinstance(rhs, IndexedElement) and rhs.rank > 0 and rhs.base.internal_variable.allocatable:
             d_lhs['allocatable'] = False
             d_lhs['is_pointer' ] = True
+            if rhs.args[0].rank != len(rhs.args) - 1 or any(isinstance(a, Slice) for a in rhs.args):
+                d_lhs['is_view' ] = True
```

Comment:
Does it make sense for an object to be both a view and a pointer?

Reply:
you can say a view is a specific case of a pointer

Reply:
a view own his shape and strides unlike a pointer

Reply:
we don't need the `is_view` anymore @saniQQ , no ?

Reply:
after we fixed the bug last time I had to use it so I can free the view when reassigning pointers